core/src/usage/type_params.rs

```diff
@@ -240,6 +242,8 @@ impl UsesTypeParams for syn::TypeParamBound {
         match *self {
             syn::TypeParamBound::Trait(ref v) => v.uses_type_params(options, type_set),
             syn::TypeParamBound::Lifetime(_) => Default::default(),
+            // non-exhaustive enum
+            _ => Default::default(),
```

Comment:
Returning `default` here feels odd, since it's going to cause us to silently miss type params in new syntax constructs.

Reply:
I changed all the occurrences of `_ => Default::default(),` that I introduced. They are now explicit panics.